2-Authorization-I/1-call-graph/README.md

```diff
@@ -221,168 +227,242 @@ In **Azure AD**, the scopes (permissions) set directly on the application regist
+                                            : null
+                                    });
+                                    break;
+                            }
+                            setResponse(tokenResponse);
+                        } catch (error) {
+                            if (error.errorCode === 'popup_window_error') {
+                                tokenResponse = await instance.acquireTokenRedirect({
+                                    scopes: scopes,
+                                    account: account,
+                                    claims: localStorage.getItem(
+                                        `cc.${msalConfig.auth.clientId}.${account.idTokenClaims.oid}`
+                                    )
+                                        ? window.atob(
+                                              localStorage.getItem(
+                                                  `cc.${msalConfig.auth.clientId}.${account.idTokenClaims.oid}`
+                                              )
+                                          )
+                                        : null
+                                });
+                                setResponse(tokenResponse);
+                            } else {
+                                setError(error);
+                            }
+                        }
+                    }
                 }
-            });
-        }
-    }, [account, instance]);
-
-    if (accounts.length > 0) {
-        return (
-            <>
-                <span>There are currently {accounts.length} users signed in!</span>
-                {apiData && (<span>Data retrieved from API: {JSON.stringify(apiData)}</span>)}
-            </>
-        );
-    } else if (inProgress === "login") {
-        return <span>Login is currently in progress!</span>
-    } else {
-        return <span>There are currently no users signed in!</span>
-    }
-}
-```
-
-If `acquireTokenSilent()` fails, the recommended pattern is to fallback to one of the interactive methods i.e. `acquireTokenPopup()` or `acquireTokenRedirect()`. In the sample, each of these options are illustrated:
+            }
+        };
+        getToken();
+    }, [account, inProgress, instance]);
 
-- [Profile.jsx](./SPA/src/pages/Profile.jsx) uses `acquireTokenSilent` and falls back to `acquireTokenPopup` if interaction is required
-- [Tenant.jsx](./SPA/src/pages/Tenant.jsx) uses `acquireTokenSilent` and falls back to `acquireTokenRedirect` if interaction is required
-- [Mails.jsx](./SPA/src/pages/Mails.jsx) uses `acquireTokenRedirect` only.
+    return [response, error];
+};
 
-> :information_source: When using `acquireTokenRedirect`, you may want to set `navigateToLoginRequestUrl` in [msalConfig](./SPA/src/authConfig.js) to **true** if you wish to return back to the page where acquireTokenRedirect was called.
+export default useTokenAcquisition;
+```
 
 ### Access Token validation
```

Comment:
Right after discussing acquiring a token, because of all the claims challenge related stuff in the code, we should get to discussing claims challenge after it. ### Access Token validation section should go after them.